tidy3d/components/tcad/boundary/charge.py

```diff
@@ -0,0 +1,52 @@
+from typing import Tuple, Union
+
+import pydantic.v1 as pd
+
+from tidy3d.components.tcad.boundary.abstract import HeatChargeBC
+from tidy3d.constants import CURRENT_DENSITY, VOLT
+
+
+class VoltageBC(HeatChargeBC):
+    """Electric potential (voltage) boundary condition.
+    Sets a potential at the specified boundary.
+    In charge simulations it also accepts an array of voltages.
+    In this case, a solution for each of these voltages will
+    be computed.
+
+    Example
+    -------
+    >>> bc1 = VoltageBC(voltage=2)
+    >>> bc2 = VoltageBC(voltage=[-1, 0, 1])
+    """
+
+    voltage: Union[pd.FiniteFloat, Tuple[pd.FiniteFloat, ...]] = pd.Field(
+        title="Voltage",
+        description="Electric potential to be applied at the specified boundary.",
+        units=VOLT,
+    )
```

Comment:
Shouldn't this now derive from `AbstractStaticTransferSourceDC` or I misunderstood?

Reply:
Is the intention to use BCs to apply the bias voltage?

Reply:
I was thinking about this more today.

We need to do the PotentialData completely differently if we want to support multiple DC + RF sources. I'm not sure the current implementation will be nicely integrate-able with different sets of analysis.

Reply:
I think I'm not following. So far this is always DC. Maybe we can have a chat? I we might be thinking about different things? :sweat_smile: